xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java

```diff
@@ -260,7 +266,7 @@ public void resolve_foundResource() {
     assertThat(result.getAddresses()).isEmpty();
     Map<String, ?> serviceConfig =
         result.getAttributes().get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-    verify(serviceConfigParser).parseServiceConfig(eq(serviceConfig));
+    assertThat(result.getServiceConfig().getConfig()).isEqualTo(serviceConfig);
```

Comment:
```
assertThat(result.getServiceConfig()).isEqualTo(
    serviceConfigParser.parseServiceConfig(serviceConfig));
```

Reply:
Wait wait. That's not correct. I use a fake `ServiceConfigParser`, which is an identity parser, returns the raw map back. So it is known that the result is equal to the original service config map. `isEqualTo(serviceConfigParser.parseServiceConfig(serviceConfig))` includes verification for the fake `ServiceConfigParser`'s implementation.

Reply:
No. `assertThat(result.getServiceConfig().getConfig()).isEqualTo(serviceConfig)` includes verification for the fake ServiceConfigParser's implementation. Because in general it does not and should not equal to `serviceConfig`, only for your particular test code parser implementation it is equal.

On the other hand `result.getServiceConfig()` is always equal to `serviceConfigParser.parseServiceConfig(serviceConfig)` regardless of parser implementation.

Reply:
Take my words back from my previous comment, that's not the case. The fake `ServiceConfigParser`'s implementation is in the recipe of producing the test outcome. However, `assertThat(result.getServiceConfig()).isEqualTo( serviceConfigParser.parseServiceConfig(serviceConfig));` is trying to cancel the effect of input in test's output. So the net outcome is nothing, we just ran the code under test once. Nothing is actually verified. This is like doing `x + 1 = x + 1`.

Reply:
> assertThat(result.getServiceConfig()).isEqualTo( serviceConfigParser.parseServiceConfig(serviceConfig)); is trying to cancel the effect of input in test's output

I don't understand the above. 

If `XdsNameResolver` has a bug and does not update the listener result, or if it updates the listener result with a wrong `ConfigOrError`, then `assertThat(result.getServiceConfig()).isEqualTo( serviceConfigParser.parseServiceConfig(serviceConfig))` will fail, but something like `x + 1 = x + 1` does not fail.

Reply:
`assertThat(result.getServiceConfig()).isEqualTo( serviceConfigParser.parseServiceConfig(serviceConfig))` makes the fake `ServiceConfigParser` code garbage.

Reply:
Why do you need fake ServiceConfigParser implementation be critical? Shouldn't its implementation a garbage implementation?